File: python.ejemplos/07.clases/03-class_vs_static.rb
class Person
  @@country = 'Spain'

  def initialize(name, age)
    @name = name
    @age = age
  end

  def to_s
    "Name: #{@name}, Age: #{@age}"
  end

  # Use class method when need access to Class state
  def self.imfrom
    puts("I'm from " + @@country)
  end

  # Use static method when dosn't need access to Class state
  def self.talk
    puts 'bla, bla, bla'
  end
end

Person.imfrom # class method. Not instance method.
              # Ruby may invoke class methods without instance
Person.talk   # class method. Not instance method.

# Python call this static method.
# Ruby static methods aren't defined into a Class.
# Ruby use modules to define methods out from Classes
module PersonTalking
  def self.talk
    puts 'bla, bla, bla from module'
  end
end

PersonTalking.talk

File: python.ejemplos/07.clases/03-class_vs_static.py
import sys

class Person:
    country = 'Spain'
    def __init__(self, name, age):
        self.name = name
        self.age = age

    def __str__(self):
        return "Name: " + self.name + ", Age: " + str(self.age)

    def imfrom(cls):
        print "I'm from " + cls.country

    @staticmethod
    def talk():
        print('bla, bla, bla')

# Person.greet() => TypeError: unbound method greet() must be called with Person instance as first argument (got nothing instead)
p = Person('Jane',0)
p.imfrom() # Python couldn't invoke class methods without instance.
Person.talk()
